Add docker-compose configuration to "Installing PufferPanel using Docker" documentation

nalvarez508 opened the pull request:
I use compose files for my docker setups as I find they're easier to manage, which is what I did for my Pufferpanel setup. Provided in this documentation update is my compose file and equivalent instructions.

Summary of changes:
- Separate installation as "command line" and "docker compose"
- Include modified instructions of CLI install in the Compose section
- Create a docker-compose example file with relevant comments to guide user

nalvarez508 committed:
Add docker-compose instructions

- define a compose file
- separate CLI vs compose

nalvarez508 committed:
Formatting code blocks, slight clarity

nalvarez508 committed:
Add a horiz. rule between two methods of install

LordRalex commented on source/installing-docker.rst:
```diff
@@ -16,9 +18,47 @@ To create the container, start it, and add the default user:
+          - 8080:8080 # XXXX:8080, where port XXXX on the outside (host) will be sent to 8080 inside the container
+          - 5657:5657 # YYYY:5657
+        volumes:
+          - ./pufferpanel-config:/etc/pufferpanel
```
You can use a docker volume here directly with compose, and thats better than another host bind.

nalvarez508 replied:
Configuration directory binding changed from host bind to a named docker volume created with the compose file.

nalvarez508 committed:
Modify volume declarations

- remove host binding for puff-config
- add a named volume for puff-config
- mapped named volume to /etc/pufferpanel

gamemodeop reviewed (approved):
Is working

LordRalex commented:
If still desired, please change the branch target to 3.x
